fuzz/client.c

```diff
@@ -55,23 +55,30 @@ int FuzzerInitialize(int *argc, char ***argv)
 
 int FuzzerTestOneInput(const uint8_t *buf, size_t len)
 {
-    SSL *client;
-    BIO *in;
-    BIO *out;
-    SSL_CTX *ctx;
+    SSL *client = NULL;
+    BIO *in = NULL;
+    BIO *out = NULL;
+    SSL_CTX *ctx = NULL;
 
     if (len == 0)
         return 0;
 
     /* This only fuzzes the initial flow from the client so far. */
     ctx = SSL_CTX_new(SSLv23_method());
+    if (ctx == NULL)
+        goto end;
 
     client = SSL_new(ctx);
+    if (client == NULL)
+        goto end;
     OPENSSL_assert(SSL_set_min_proto_version(client, 0) == 1);
     OPENSSL_assert(SSL_set_cipher_list(client, "ALL:eNULL:@SECLEVEL=0") == 1);
     SSL_set_tlsext_host_name(client, "localhost");
     in = BIO_new(BIO_s_mem());
     out = BIO_new(BIO_s_mem());
+    if (in == NULL || out == NULL) {
+	goto end;
```

Comment:
Wrong indentation. You also need to BIO_free() the in and out bios as one of them might be non-NULL.

Reply:
in = BIO_new(BIO_s_mem());
+    if (in == NULL) {
+        goto end;
+    }
     out = BIO_new(BIO_s_mem());
-    if (in == NULL || out == NULL) {
+    if (out == NULL) {
+        BIO_free(in);
        goto end;
     }

Is this ok?

Reply:
I'm assuming you mean no braces are needed?

Reply:
Braces are needed, but the goto statement is not indented properly.